contracts/core/EntryPoint.sol

```diff
@@ -284,12 +268,13 @@ contract EntryPoint is IEntryPoint, StakeManager {
     }
 
     /**
-     * helper: make a "view" call to calculate the sender address.
-     * must be called from zero-address.
+     * Get counterfactual sender address.
```

Comment:
What was the rationale for returning the address in an error?

Reply:
This "getter" has side effect (creating the wallet), so it has to revert to make sure no one attempt to call it on chain. Yes, we're rewriting it as a function returning address wrapping an internal function that always reverts...